emukit/core/interfaces/models.py

```diff
@@ -74,16 +74,15 @@ def get_joint_prediction_gradients(self, X: np.ndarray) -> Tuple[np.ndarray, np.
 
 class IPriorHyperparameters:
     def generate_hyperparameters_samples(self, n_samples: int, n_burnin: int,
-                                         subsample_interval: int, step_size: float, leapfrog_steps: int) -> np.ndarray:
+                                         subsample_interval: int, step_size: float, leapfrog_steps: int) -> None:
         """
-        Generates the samples from the hyper-parameters of the model.
+        Generates the samples from the hyper-parameters of the model, and sets self.samples to that
```

Comment:
I quite like the stateless nature of this method, I think it makes things explicit. As commented above, IntegratedAcquisition seem to be the only place that actually needs to keep samples as its state.

Reply:
We'll investigate storing the samples in `IntegratedHyperParameterAcquisition` instead of the model.

Reply:
Now done.